docs/topics/downloader-middleware.rst

```diff
@@ -922,13 +922,72 @@ RobotsTxtMiddleware
     To make sure Scrapy respects robots.txt make sure the middleware is enabled
     and the :setting:`ROBOTSTXT_OBEY` setting is enabled.
 
+    This middleware has to be combined with a `robots.txt <http://www.robotstxt.org/>`_ parser.
+
+    Scrapy ships with support for three `robots.txt <http://www.robotstxt.org/>`_ parsers:
+
+    * `RobotFileParser`_
+    * `Reppy`_
+    * `Robotexclusionrulesparser`_
+
+    You can change the `robots.txt <http://www.robotstxt.org/>`_ parser with the :setting:`ROBOTSTXT_PARSER`
+    setting. Or you can also :ref:`implement support for a new parser <support-for-new-robots-parser>`.
+
 .. reqmeta:: dont_obey_robotstxt
 
 If :attr:`Request.meta <scrapy.http.Request.meta>` has
 ``dont_obey_robotstxt`` key set to True
 the request will be ignored by this middleware even if
 :setting:`ROBOTSTXT_OBEY` is enabled.
 
+.. _support-for-new-robots-parser:
+
+Implementing support for a new parser
+-------------------------------------
+
+You can implement support for a new `robots.txt <http://www.robotstxt.org/>`_ parser by implementing the
+interface described below.
+
+.. module:: scrapy.extensions.robotstxtparser
+
+.. class:: RobotsParser
+
+    .. method:: __init__(content)
+
+      Parse content of `robots.txt <http://www.robotstxt.org/>`_ file.
+
+      :param content: Content of `robots.txt <http://www.robotstxt.org/>`_ file. For Python 2.7, if ``content``
+                    contains non-ASCII characters, it should be ``UTF-8`` encoded. 
```

Comment:
I think the content type should be bytes (encoded string), since some of the parsers may work with bytes and for those that don’t parser classes can simply decode those bytes first.

Also, unlike with the documentation of an API that developers _use_, when you document an interface that developers _implement_, parameters are documented how they _are_ and return values how they _must be_. That is, you should not say ‘it should be’, you should say ‘it is’.

In this specific case, however, I would actually refactor the sentence a bit further. If we are passing the file contents as bytes (`response.body` from Scrapy), we should do it regardless of Python version. So this may be enough:

    Content of `robots.txt <http://www.robotstxt.org/>`_ file as bytes.

You don’t need to mention the encoding. Since Scrapy is simply going to pass the received content _as is_, we can not actually enforce a specific encoding, so the encoding really depends on the website serving the file. You can add a note that explains that robots.txt files are expected to be encoded in UTF-8, but you should not suggest that Scrapy ensures this in any way.